src/wasm-interpreter.h

```diff
@@ -1130,8 +1133,20 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
   }
   Flow visitDrop(Drop* curr) {
     NOTE_ENTER("Drop");
+    seenUnknownLocalGet = false;
     Flow value = visit(curr->value);
     if (value.breaking()) {
+      // Handle the case where all we don't know to perform the drop is a local
+      // value, which we do not need to know if there are no other side-effects.
+      // Doesn't apply to globals since these would error when module is given.
+      if (seenUnknownLocalGet && value.breakTo == NONCONSTANT_FLOW &&
+          module != nullptr) {
```

Comment:
I didn't review #2702 and I just noticed that `Flow.breaking()` returns true not only when we are actually breaking or returning, but also every time we can't precompute a value. I'm not sure if this is the design you intended, but it feels confusing..?

Reply:
Idea is that we do a cheap check on `seenUnknownLocalGet` first so we only spawn the analyzer if there is the chance that we succeed, but the flow might still be breaking to a label, which will never succeed. Except when breaking to `NONCONSTANT_FLOW`, which is the expected break target when `seenUnknownLocalGet` is `true`.

Reply:
What I meant is, I find `Flow.breaking()` returns true on non-breaking flows a little confusing. Before `breaking()` returns true only for control-flow-transferring expressions, such as branches or returns. But now it returns true too when it is set to `NONCONSTANT_FLOW`, which means every time you can't precompute a value.

Reply:
Yeah, this mechanism existed prior to my PRs, with two special break targets `RETURN_FLOW` and `NONCONSTANT_FLOW` (previously `NONSTANDALONE_FLOW` in precompute, then generalized and moved) indicating special conditions. My impression was that this is done so we don't need to throw exceptions, but can reuse `breakTo`?

Reply:
The mechanism predates your PR is confined to `PrecomputeExpressionRunner`, and it [handles](https://github.com/WebAssembly/binaryen/blob/50fee5228acfbc3ffb882c5a1489475589d9fc1f/src/passes/Precompute.cpp#L173-L175) it specially when checking `Flow.breaking()`. But now after #2702, all classes that inherits from `ExpressionRunner` treats `NONCONSTANT_FLOW` as breaking flow. It will return true every time we call `Flow.breaking()`, even if it's not breaking. And it is not practical to add checks like [this](https://github.com/WebAssembly/binaryen/blob/50fee5228acfbc3ffb882c5a1489475589d9fc1f/src/passes/Precompute.cpp#L173-L175) every time we call `Flow.breaking()`.

Reply:
From looking at the invocations of `Flow.breaking()`, these do either of the following:

* If the flow is breaking, maybe clean up and return the breaking flow, letting the caller deal with it or
* If the flow is breaking, check that `breakTo` is the expected label and error otherwise or
* If the flow is not breaking, do whatever

In all these cases whether `breakTo` is `NONCONSTANT_FLOW` is irrelevant, quite similar to `RETURN_FLOW`, which is based on the same assumption.

Reply:
Hmm, I guess my main complaint is after all related to #2797. So I think `Flow.breaking()` should return true only when it is actually breaking, i.e., control flow transferring. But as you said, there was an exception of `PrecomputeExpressionRunner`, in which we had `NOTSTANDALONE_FLOW` or something. Now we effectively merged `ExpressionRunner` and `PrecomputeExpressionRunner`, the distinction is not clear, and suddenly we are treating nonbreaking flows as breaking. I agree this may not be a problem from the user perspective, but I find this pretty confusing now. Basically it feels like, we moved most of functionalities of `PrecomputeExpressionRunner` into `ExpressionRunner` base class even though many of them don't apply to other subclass `RuntimeExpressionRunner`, and `RuntimeExpressionRunner` is overriding all them. I guess we should continue this discussion in #2797.